pkg/client/didexchange/client.go

```diff
@@ -19,11 +21,13 @@ import (
 // provider contains dependencies for the DID exchange protocol and is typically created by using aries.Context()
 type provider interface {
 	Service(id string) (interface{}, error)
+	Wallet() wallet.Wallet
```

Comment:
We might want to consider having a subset of the Wallet interface.

E.g., I'm guessing the Client doesn't call Open and Close.

Reply:
yes

Reply:
I will work into break it into multiple interfaces

Reply:
done